core/01_core.js

```diff
@@ -133,13 +133,48 @@
     return promiseRing[idx] != NO_PROMISE;
   }
 
-  function opresolve() {
-    for (let i = 0; i < arguments.length; i += 2) {
+  const macrotaskCallbacks = [];
+  const nextTickCallbacks = [];
+
+  function setMacrotaskCallback(cb) {
+    ArrayPrototypePush(macrotaskCallbacks, cb);
+  }
+
+  function setNextTickCallback(cb) {
+    ArrayPrototypePush(nextTickCallbacks, cb);
+  }
+
+  // This function has variable number of arguments. The last argument describes
+  // if there's a "next tick" scheduled by the Node.js compat layer. Arguments
+  // before last are alternating integers and any values that describe the
+  // responses of async ops.
+  function eventLoopTick() {
```

Comment:
In a future patch (definitely not this one) it might be worth evaluating the cost of using `arguments` here vs an array. Does v8 still deoptimize functions that use arguments?

Reply:
Good point, I can do a quick benchmark after this PR lands to check it.

Reply:
Quick benchmark shows about 1.5k RPS improvement in `Deno.serve()` bench. I will open a PR for that change and would be happy to merge it and evaluate on real world benchmarks.